src/plugins/dataflow/model/dataflow-content.ts

```diff
@@ -227,36 +222,43 @@ export const DataflowContentModel = TileContentModel
-        const tableTile = getTileContentById(self, tableId); //get tableTile contents given a tableId
+        const tableTileContents = getTileContentById(self, tableId); //get tableTile contents given a tableId
+        const tableSharedModels = sharedModelManager.getTileSharedModels(tableTileContents);
+        if (tableSharedModels.length > 1){ //table ideally should only have 1 shared dataSet
+          console.warn("Table has more than one shared dataSet");
+        }
         //sever connection table -> table sharedDataSet
-        sharedTable && sharedModelManager.removeTileSharedModel(tableTile, sharedTable);
+        tableSharedModels && sharedModelManager.removeTileSharedModel(tableTileContents, tableSharedModels[0]);
         //connect table -> dataflow sharedDataset
-        self.sharedModel && sharedModelManager.addTileSharedModel(tableTile, self.sharedModel);
+        self.sharedModel && sharedModelManager.addTileSharedModel(tableTileContents, self.sharedModel);
       }
       else {
-        console.warn("GeometryContent.addLinkedTable unable to link table");
+        console.warn("DataflowContent.addLinkedTable unable to link table");
       }
     },
     removeLinkedTable(tableId: string) {
       const sharedModelManager = self.tileEnv?.sharedModelManager;
       if (sharedModelManager?.isReady && self.isLinkedToTable(tableId)) {
-        const sharedTable = sharedModelManager.findFirstSharedModelByType(SharedDataSet, tableId);
         //sever connection table -> table sharedDataSet
         const tableTileContents = getTileContentById(self, tableId); //get tableTile contents given a tableId
-        sharedTable && sharedModelManager.removeTileSharedModel(tableTileContents, sharedTable);
+        self.sharedModel && sharedModelManager.removeTileSharedModel(tableTileContents, self.sharedModel);
         //create a dataSet with two attributes with X / Y, link table tile to this dataSet
         const title = tableTileContents ? getTileTitleFromContent(tableTileContents) : undefined;
-        const dataSet = DataSet.create({name: title});
-        addAttributeToDataSet(dataSet, { name: "x" });
-        addAttributeToDataSet(dataSet, { name: "y" });
-        const newSharedDataSet = SharedDataSet.create({ providerId: tableId, dataSet });
+        const newDataSet = this.createDefaultDataSet(title);
+        const newSharedDataSet = newDataSet && SharedDataSet.create({ providerId: tableId, dataSet: newDataSet });
         sharedModelManager.addTileSharedModel(tableTileContents, newSharedDataSet);
       }
       else {
-        console.warn("GeometryContent.addLinkedTable unable to unlink table");
+        console.warn("DataflowContent.addLinkedTable unable to unlink table");
       }
+    },
+    createDefaultDataSet(title: string | undefined){
+      const dataSet = DataSet.create({name: title});
+      addAttributeToDataSet(dataSet, { name: "x" });
+      addAttributeToDataSet(dataSet, { name: "y" });
+      return dataSet;
```

Comment:
I think @tealefristoe was suggesting to put this function in the DataSet code not here. 
It could go in `models/data/data-set` or `models/shared/shared-data-set`.

Reply:
Created a file `create-default-data-set.ts` inside of `plugins/dataflow/models/utilities/`